src/test/java/gov/healthit/chpl/aqa/stepDefinitions/UploadListingsRegularlySteps.java

```diff
@@ -144,8 +136,19 @@ public void confirmUploadedListing(final String edition, final String testChplId
         System.out.println("Confirmed");
 
         WebDriverWait wait = new WebDriverWait(driver, LONG_TIMEOUT);
-        wait.until(ExpectedConditions.visibilityOf(DpManagementPage.updateSuccessfulToastContainerText(driver)));
+        wait.until(ExpectedConditions.visibilityOf(DpManagementPage.updateSuccessfulToastContainer(driver)));
+        } catch (Exception e) {
+            takeScreenshot();
+        }
+    }
 
+    /**
+     * Take a screenshot.
+     * @throws Exception if there is an exception
+     */
+    public void takeScreenshot() throws Exception {
```

Comment:
Good start. We'll likely want this function to exist in all of the Step files, so would recommend moving it to a "BaseStep.java" class and have at least this step file extend that base file.

Also, should do something to make the filename more unique; if more than one screenshot is taken during a run they'll overwrite each other. Could append the timestamp, with something like `filePath + File.separator = "failed-test-" + (new Date()).getTime() + ".png"`

Reply:
I have it in my 'TO-DO' list :)